internal/stores/hostdb.go

```diff
@@ -314,10 +314,10 @@ func (db *SQLStore) Hosts(notSince time.Time, max int) ([]hostdb.Host, error) {
 	// given time.
 	var foundHosts [][]byte
 	err := db.db.Table("hosts").
-		Joins("JOIN host_interactions ON host_interactions.db_host_id = hosts.ID").
+		Joins("LEFT JOIN host_interactions ON host_interactions.db_host_id = hosts.ID").
 		Select("Public_key").
 		Group("Public_Key").
-		Having("MAX(Timestamp) < ?", notSince.UTC()). // use UTC since we stored timestamps in UTC
+		Having("MAX(COALESCE(Timestamp, CAST(0 as datetime))) < ?", notSince.UTC()). // use UTC since we stored timestamps in UTC
```

Comment:
Have you tried

`Having("IsNull(MAX(Timestamp), CAST(0 as datetime)) < ?", notSince.UTC()).` as well?

Feels a bit less hacky than using `COALESCE` on an array with an added default value and also a bit faster since we only look at the end result and convert that if it's null.

Reply:
so I actually tried `ISNULL` but my query editor threw a warning that put me on the wrong foot and I just went with `COALESCE` - `IFNULL` works though 👍  extended the test to be doubly sure